src/hash_http_content/hasher.py

```diff
@@ -168,18 +169,37 @@ def _handle_html(self, contents: bytes, encoding: str) -> HandlerResult:
         logging.debug("Handling content as HTML")
         self.__init_browser()
 
-        if encoding:
-            html = str(contents, encoding)
-        else:
-            html = str(contents, self._default_encoding)
+        # Until the Page.setContent() method allows options, writing the HTML
+        # document to a temporary file and navigating to it with Page.goto() is
+        # the only way to leverage the `waitUntil` option to give time for the
+        # page's contents to load. Support for options in Page.setContent() is
+        # expected in pyppeteer when the puppeteer v2.1.1 feature parity rewrite
+        # is completed per:
+        # https://github.com/pyppeteer/pyppeteer/issues/134 for more information
+        with tempfile.NamedTemporaryFile(suffix=".html") as fp:
+            # Output to a temporary file so it's available to the browser
+            fp.write(contents)
+            fp.flush()
+
+            page: Page = asyncio.get_event_loop().run_until_complete(
+                self._browser.newPage()
+            )
+
+            logging.debug("Navigating to temporary file '%s'", fp.name)
+            # Wait for everything to load after navigating to the temporary file
+            asyncio.get_event_loop().run_until_complete(
+                page.goto(f"file://{fp.name}", {"waitUntil": ["load", "networkidle2"]})
+            )
+            page_contents: str = asyncio.get_event_loop().run_until_complete(
+                page.content()
+            )
+
+            asyncio.get_event_loop().run_until_complete(page.close())
 
-        logging.debug("Setting page contents and rendering")
-        page: Page = asyncio.get_event_loop().run_until_complete(
-            self._browser.newPage()
+        # Try to guarantee our preferred encoding
+        page_contents = bytes(page_contents.encode(self._default_encoding)).decode(
```

Comment:
Should this line have a type hint?

Reply:
Certainly wouldn't hurt. I get bad about it sometimes when a variable is only assigned to once from a "guaranteed output". I'll go back and add some more type hinting to keep it all fresh and statically happy (especially since I added new instance variables with the expectation that they can be changed).

Reply:
@jsf9k That is actually hinted here (just above the line you were looking at):
https://github.com/cisagov/hash-http-content/blob/2c51f2b29dc9971127572d86dc089a5947a6d5f9/src/hash_http_content/hasher.py#L216
, but I took care of some low-hanging/should be there type hints in 2c51f2b.